pkg/wallet/internal/mapping/mapping_list_actions.go

```diff
@@ -0,0 +1,187 @@
+package mapping
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk/primitives"
+	"github.com/bsv-blockchain/go-sdk/chainhash"
+	"github.com/bsv-blockchain/go-sdk/script"
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	sdk "github.com/bsv-blockchain/go-sdk/wallet"
+	"github.com/go-softwarelab/common/pkg/slices"
+)
+
+// MapListActionsArgs maps sdk.ListActionsArgs to wdk.ListActionsArgs
+func MapListActionsArgs(args sdk.ListActionsArgs) wdk.ListActionsArgs {
+	result := wdk.ListActionsArgs{
+		Labels: slices.Map(args.Labels, func(label string) primitives.StringUnder300 { return primitives.StringUnder300(label) }),
+		Limit:  primitives.PositiveIntegerDefault10Max10000(args.Limit),
+		Offset: primitives.PositiveIntegerDefault10Max10000(args.Offset),
+	}
+
+	switch args.LabelQueryMode {
+	case sdk.QueryModeAll:
+		labelQueryMode := primitives.LabelQueryModeString("all")
+		result.LabelQueryMode = &labelQueryMode
+	case sdk.QueryModeAny:
+		labelQueryMode := primitives.LabelQueryModeString("any")
+		result.LabelQueryMode = &labelQueryMode
+	default:
+		labelQueryMode := primitives.LabelQueryModeString("any")
+		result.LabelQueryMode = &labelQueryMode
+	}
+
+	if args.SeekPermission != nil {
+		seekPermission := primitives.BooleanDefaultTrue(*args.SeekPermission)
+		result.SeekPermissions = &seekPermission
```

Comment:
small simplification for this and all the next.
```suggestion
		result.SeekPermissions =  to.Ptr(primitives.BooleanDefaultTrue(*args.SeekPermission))
```